tests/examples/requirements-pip-dict/dependencies.yaml

```diff
@@ -0,0 +1,14 @@
+files:
+  all_of_the_things:
+    output: requirements
+    includes:
+      - run_deps
+    requirements_dir: .
+dependencies:
+  run_deps:
+    common:
+      - output_types: [requirements]
+        packages:
+          - fsspec>=0.6.0
+          - pip:
+            - pandas<1.0
```

Comment:
Before this PR, doing something like this would result in an error that'd require understanding the internals of `rapids-dependency-file-generator`.

```text
Traceback (most recent call last):
  File "/Users/jlamb/miniforge3/envs/dfg-dev/bin/rapids-dependency-file-generator", line 8, in <module>
    sys.exit(main())
  File "/Users/jlamb/repos/dependency-file-generator/src/rapids_dependency_file_generator/_cli.py", line 151, in main
    make_dependency_files(
  File "/Users/jlamb/repos/dependency-file-generator/src/rapids_dependency_file_generator/_rapids_dependency_file_generator.py", line 428, in make_dependency_files
    contents = make_dependency_file(
  File "/Users/jlamb/repos/dependency-file-generator/src/rapids_dependency_file_generator/_rapids_dependency_file_generator.py", line 146, in make_dependency_file
    file_contents += "\n".join(dependencies) + "\n"
TypeError: sequence item 1: expected str instance, dict found
```

As of this PR, I think the error message `rapids-dependency-file-generator` will point people pretty directly to the issue.

```text
Traceback (most recent call last):
  File "/Users/jlamb/miniforge3/envs/dfg-dev/bin/rapids-dependency-file-generator", line 8, in <module>
    sys.exit(main())
  File "/Users/jlamb/repos/dependency-file-generator/src/rapids_dependency_file_generator/_cli.py", line 151, in main
    make_dependency_files(
  File "/Users/jlamb/repos/dependency-file-generator/src/rapids_dependency_file_generator/_rapids_dependency_file_generator.py", line 432, in make_dependency_files
    contents = make_dependency_file(
  File "/Users/jlamb/repos/dependency-file-generator/src/rapids_dependency_file_generator/_rapids_dependency_file_generator.py", line 148, in make_dependency_file
    raise ValueError(f"Map inputs like {dep} are not allowed for the 'requirements' file type.")
ValueError: Map inputs like {'pip': ['pandas<1.0']} are not allowed for the 'requirements' file type.
```